### DISCUSSION TITLE
Unify sync options

### DISCUSSION THREAD

**Discussion by ph00lt0** [ente-io/ente#610] [Enhancements]
Currently I see two sync options. 

- Export folder 
   syncs all data from ente to device. Changes on device however are not impacting on ente
- watch folders 
   syncs data from folder to ente.

What I want:
One folder on machine that has all photos from ente and changes made there are synced to ente.

**Comment by vishnukvmd** [comment]
This will unfortunately be difficult to implement in the short term, since we plan to enhance our export flow by embedding extra metadata back into files where ever possible (#202, ente-io/ente#607). Which will make it difficult to provide a Dropbox-like sync, since the files on your disk will now be an "enhanced" version of the files on the server.

That said, thank you for sharing your use case with us. We'd love to offer a simple 2-way sync in the future, so will keep this issue open until we ship this.

**Comment by ph00lt0** [comment]
First really have to say that i admire your engagement with the community.

I understand this isn't a quick fix and I wasn't expecting that either. 

Having thought about this a bit more. I do have another temporary suggestion that fixes the main issue for me here. 

So right now, i have a folder structure with all my photos in Nextcloud in nested folders, whom are shared with various people. I doubt I can just drag and drop this in ente. And i think an import feature then would be great. If i would use the folder watch option I won't be able to delete the origin afterwards as far as I understand so that wouldn't work. 

I am convinced on your product and will definitely migrate and get subscribed soon. I just got to figure out the best migration strategy.

**Comment by vishnukvmd** [comment]
We unfortunately don't support nested albums (yet). So in this case the desktop app will flatten the hierarchy.

If you're okay with that, drag-and-drop will work well. Even if the upload breaks (due to an app / system / network issue), the app will resume from where it left off.

> I won't be able to delete the origin afterwards

If this is breaking your workflow, you could consider using our desktop app's export feature to "continuously sync" data to a folder that Nextcloud is watching (making Nextcloud a replica). Now  if you delete files from ente, it will remove them from your local disk (and subsequently Nextcloud) as well.

Would that help?

**Comment by ph00lt0** [comment]
Hey thanks for your reply again.

I don't want to keep nextcloud. I am really happy to have finally found a replacement for it. But yeah that actually seems to be a good option for people who would want that.

I don't really mind to lose the by year folders that's fine but I was afraid drag and drop would indeed not work for the amount of data I will have to sync up. But if you say that this shouldn't be an issue I will definitely give that a try. 

Thank you so much for your dedication and support.

**Comment by ph00lt0** [comment]
Just to report back here that this actually went really smooth and I was very happy with the auto continue. It is solid. 

But one more request about the export folder. Would it be possible to somehow mark that as read only? Because that would make it more clear that one cannot modify data there in the file managers.

**Comment by vishnukvmd** [comment]
> Would it be possible to somehow mark that as read only?

Unsure about the side effects this may have. Say a user wants to delete a large, unimportant album to save disk space, this would get in the way.

**Comment by ph00lt0** [comment]
I think i would make sense to choose which albums to sync. But by removing them you kinda lose control over that too. I just feel like it would be very confusing as the situation isn't really in sync.

**Comment by vishnukvmd** [comment]
Giving an option to choose the albums to sync makes a lot of sense.

We're currently caught up in implementing client-side search, will circle back and ship these quality-of-life improvements right after.

**Comment by rollermy** [comment]
I'd like to sync the pictures with my desktop too. 
The desktop app does not seem to cache all the pictures either, so displaying them takes a while on my slow internet.

**Comment by vishnukvmd** [comment]
Hey @rollermy, just so we're on the same page, we support exporting data to your a folder (via the `Export Data > Sync continuously` option).

We are also tracking improvements for an offline experience @ https://github.com/ente-io/photos-desktop/issues/231.

**Comment by rollermy** [comment]
Thank you. ente-io/photos-desktop#231 is what i'm looking for.

**Comment by pixel-fabian** [comment]
I came here because of a question regarding the sync options on [mastodon](https://chaos.social/@pixel_fabian/112927903194024792) and like to elaborate my use-case to see if it fits the current roadmap plans.

I use ente on multiple devices (2x desktop and 1x mobile) and it works great for the backup of my mobile pictures. But i also use a different camera, edit these pictures on desktop or laptop and want to sync them to the mobile device.
So i'd love the possibility to define a album which has 2-way sync across multiple devices. Support of sub-folders would be the icing on the cake.

**Comment by radh21301** [comment]
I know you must hear people say this a lot - But honestly, this is one major factor that prevents me from experiencing ente to its potential :)

I had a horrible time when I transitioned out of google photos - with its metadata borked up. I had to manually fix lots of dates, and GPS coordinates etc. So, what I am looking for is a way for ente to allow me to sync images from my phone. Then mount all of my ente assets on my nas/laptop - use tools like Digikam, Exiftool etc. to fix metadata = Dates, GPS etc (I have lots of images without these metatada -> screenshots, whatsapp images etc.). And then have these changes reflect across the ente ecosystem. 

So, if the day where I leave ente comes, I can be assured that all my images have the right metadata.

P.S. I know ente allows us to edit metadata, but I would rather to do it myself since I had a bad experience before.

**Comment by reitzig** [comment]
Just to add a use case: use ente for syncing _to_ a long-term storage (not _as_ long-term storage).

Specifically:

1. I take a photo with my phone.
2. Tool automatically backs up photo in cloud.
3. Tool automatically downloads photo from cloud to local folder on a PC.
4. I (on the PC) file the photo away on my local NAS.
5. I delete the photo in the local folder (PC).
6. Tool automatically deletes photo in the cloud.
7. _Optional_: Tool automatically deletes photo on the phone.

I have been running 1-6 with a tool other than ente for years. The overall UX is nicer for photos with ente, so I would like to switch. Unfortunately, it seems that 6 is missing which is kind of crucial. Also, 3 doesn't seem to be possible for individual albums ("Export Data" seems to work an the full account); I want to treat unsorted phone photos in this way, not any albums I created for sharing.
